simulationworkflowschema/general.py

```diff
@@ -231,24 +285,261 @@ def normalize(self, archive, logger):
+        description="""
+        Reference to the starting point (XC functional or HF) used.
+        """,
+    )
+
+    electrons_representation = Quantity(
+        type=Reference(BasisSetContainer),
+        description="""
+        Reference to the basis set used.
+        """,
+    )
+
+
+class DFTOutputs(SimulationWorkflowResults):
+    """
+    Base class defining the typical output properties of a DFT SinglePoint calculation.
+    """
+
+    band_gap_dft = Quantity(
+        type=Reference(BandGap),
+        shape=["*"],
+        description="""
+        Reference to the DFT band gap.
+        """,
+    )
+
+    dos_dft = Quantity(
+        type=Reference(Dos),
+        shape=["*"],
+        description="""
+        Reference to the DFT density of states.
+        """,
+    )
+
+    band_structure_dft = Quantity(
+        type=Reference(BandStructure),
+        shape=["*"],
+        description="""
+        Reference to the DFT band structure.
+        """,
+    )
+
+
+class GWOutputs(SimulationWorkflowResults):
+    """
+    Base class defining the typical output properties of a GW SinglePoint calculation.
+    """
+
+    band_gap_gw = Quantity(
+        type=Reference(BandGap),
+        shape=["*"],
+        description="""
+        Reference to the GW band gap.
+        """,
+    )
+
+    dos_gw = Quantity(
+        type=Reference(Dos),
+        shape=["*"],
+        description="""
+        Reference to the GW density of states.
+        """,
+    )
+
+    band_structure_gw = Quantity(
+        type=Reference(BandStructure),
+        shape=["*"],
+        description="""
+        Reference to the GW band structure.
+        """,
+    )
+
+
+class TBOutputs(SimulationWorkflowResults):
+    """
+    Base class defining the typical output properties of a tight-binding (TB) SinglePoint
+    calculation.
+    """
+
+    band_gap_tb = Quantity(
+        type=Reference(BandGap),
+        shape=["*"],
+        description="""
+        Reference to the TB band gap.
+        """,
+    )
+
+    dos_tb = Quantity(
+        type=Reference(Dos),
+        shape=["*"],
+        description="""
+        Reference to the TB density of states.
+        """,
+    )
+
+    band_structure_tb = Quantity(
+        type=Reference(BandStructure),
+        shape=["*"],
+        description="""
+        Reference to the TB band structure.
+        """,
+    )
+
+
+class DMFTOutputs(SimulationWorkflowResults):
+    """
+    Base class defining the typical output properties of a DMFT SinglePoint calculation.
+    """
+
+    band_gap_dmft = Quantity(
+        type=Reference(BandGap),
+        shape=["*"],
+        description="""
+        Reference to the DMFT band gap.
+        """,
+    )
+
+    greens_functions_dmft = Quantity(
+        type=Reference(GreensFunctions),
+        shape=["*"],
+        description="""
+        Ref to the DMFT Greens functions.
+        """,
+    )
+
+
+class MaxEntOutputs(SimulationWorkflowResults):
```

Comment:
should this be in max_ent.py

Reply:
See the comment above, but basically, what about other possible workflows that use MaxEntOutputs as well? I am trying to come up with some base class definitions that might be shareable with several workflows.

I am wondering whether the normalization of extracting results should be in these classes, and not in the ones for the workflow.